front/src/modules/trainschedule/components/Timetable/TimetableToolbar.tsx

```diff
@@ -71,65 +72,93 @@ const TimetableToolbar = ({
     useState<ScheduledPointsHonoredFilter>('both');
   const [selectedTags, setSelectedTags] = useState<Set<string | null>>(new Set());
 
+  const [totalPacedTrainCount, totalTrainScheduleCount] = timetableItems.reduce(
+    (acc, { id }) => {
+      if (isTrainSchedule(id)) {
+        acc[1] += 1;
+      } else {
+        acc[0] += 1;
+      }
+      return acc;
+    },
+    [0, 0]
+  );
+
   const debouncedFilter = useDebounce(filter, 500);
 
   const debouncedRollingstockFilter = useDebounce(rollingStockFilter, 500);
 
   const [deleteTrainSchedules] = osrdEditoastApi.endpoints.deleteTrainSchedule.useMutation();
 
+  const { trainScheduleIds, pacedTrainIds } = selectedTimetableItemIdsByType;
+  const selectedTimetableItemIds = [...trainScheduleIds, ...pacedTrainIds];
```

Comment:
Since selectedTimetableItems now contain all the item ids flattened, you need to recompute the selectedTrainScheduleIds and the selectedPacedTrainIds only here, since they are not needed the other places (selectedTimetableItemIdsByType should replace selectedTimetableItemIds):
```suggestion
   const { selectedTrainScheduleIds, selectedPacedTrainIds } = useMemo(
    () =>
      selectedTimetableItemIdsByType.reduce(
        (acc, timetableItemId) => {
          if (isTrainSchedule(timetableItemId)) {
            acc.selectedTrainScheduleIds.push(timetableItemId);
          } else {
            acc.selectedPacedTrainIds.push(timetableItemId);
          }
          return acc;
        },
        { selectedTrainScheduleIds: [], selectedPacedTrainIds: [] } as {
          selectedTrainScheduleIds: TrainScheduleId[];
          selectedPacedTrainIds: PacedTrainId[];
        }
      ),
    [selectedTimetableItemIdsByType]
  );
```

Reply:
Done